### ISSUE TITLE
[rllib] MARWIL tuned cartpole example (and my own experiments) produce nan rewards only.

### ISSUE THREAD

**Issue by maxpumperla** [ray-project/ray#9402]
### What is the problem? + Reproduction

I have a custom example that produces offline data and picks it up with MARWIL for training. I observed that I get `nan` reward values for my example every time, so I went a step back and used your cartpole example:

https://github.com/ray-project/ray/blob/cd5a207d69cdaf05b47d956c18e89d928585eec7/rllib/tuned_examples/marwil/cartpole-marwil.yaml

I'm following the exact steps there, i.e. first run 

```
./train.py --run=PPO --env=CartPole-v0 \
    --stop='{"timesteps_total": 50000}' \
    --config='{"output": "/tmp/out", "batch_mode": "complete_episodes"}'
```

followed by 

```
rllib train -f cartpole-marwil.yaml
```

I did this both on my currently preferred stable version `0.8.5`, as well as on the `0.9.0.dev0` wheel. The result is this:

```
== Status ==
Memory usage on this node: 19.4/32.0 GiB
Using FIFO scheduling algorithm.
Resources requested: 0/12 CPUs, 0/0 GPUs, 0.0/9.96 GiB heap, 0.0/3.42 GiB objects
Result logdir: /Users/maxpumperla/ray_results/cartpole-marwil
Number of trials: 2 (2 TERMINATED)
+--------------------------------+------------+-------+--------+--------+------------------+--------+----------+
| Trial name                     | status     | loc   |   beta |   iter |   total time (s) |     ts |   reward |
|--------------------------------+------------+-------+--------+--------+------------------+--------+----------|
| MARWIL_CartPole-v0_7af06_00000 | TERMINATED |       |      0 |   2206 |          58.5661 | 500007 |      nan |
| MARWIL_CartPole-v0_7af06_00001 | TERMINATED |       |      1 |   2248 |          58.6117 | 500286 |      nan |
+--------------------------------+------------+-------+--------+--------+------------------+--------+----------+
```

Also, I've noticed that your MARWIL unit test is a pure smoke test and doesn't check reward values, but I didn't run that locally. Maybe it produces nan values as well.

In any case I'd appreciate any input here, as we'd love to use MARWIL for our "real" use case, in which we see the same behaviour.

**Comment by sven1977** [comment]
Hi @maxpumperla , thanks for filing this. Yes, we don't actually have an actual MARWIL learning test, that's true. I'll take a look. I remember that the NaNs are b/c of the offline learning setup, but I'll confirm that this is a non-issue and get back to you. Either way, we shouldn't be displaying NaN and instead evaluate on a live env and report actual rewards.

**Comment by sven1977** [comment]
Which ray version are you on? I think we fixed this some time ago:
```
    def test_marwil_compilation(self):
        """Test whether a MARWILTrainer can be built with all frameworks."""
        config = marwil.DEFAULT_CONFIG.copy()
        config["num_workers"] = 0  # Run locally.

        # Test for all frameworks.
        for _ in framework_iterator(config, frameworks=("tf")):
            ray.tune.run("MARWIL", config=dict(config, **{"env": "CartPole-v0"}), verbose=1)
```

... results in ...
```
== Status ==
Memory usage on this node: 12.4/24.0 GiB
Using FIFO scheduling algorithm.
Resources requested: 1/8 CPUs, 0/0 GPUs, 0.0/7.32 GiB heap, 0.0/2.54 GiB objects
Result logdir: C:\Users\svenm\ray_results\MARWIL
Number of trials: 1 (1 RUNNING)
+--------------------------------+----------+--------------------+--------+------------------+-------+----------+
| Trial name                     | status   | loc                |   iter |   total time (s) |    ts |   reward |
|--------------------------------+----------+--------------------+--------+------------------+-------+----------|
| MARWIL_CartPole-v0_968d1_00000 | RUNNING  | 192.168.2.107:7312 |     73 |          13.2281 | 16642 |    56.09 |
+--------------------------------+----------+--------------------+--------+------------------+-------+----------+

```

**Comment by sven1977** [comment]
I'll try with an input file ...

**Comment by sven1977** [comment]
Ok, confirmed. All NaNs. But makes sense, if input is a file, then there are no samplers, hence no actual env. We should probably have sampling for the evaluation workers, though to indicate actual learning progress. ...

**Comment by sven1977** [comment]
Got it: Alter the rllib/agents/marwil/tests/test_marwil.py file as follows:

```
    def test_marwil_compilation(self):
        """Test whether a MARWILTrainer can be built with all frameworks."""
        config = marwil.DEFAULT_CONFIG.copy()
        config["num_workers"] = 0  # Run locally.
        config["evaluation_num_workers"] = 1
        config["evaluation_interval"] = 1  # <- this is switched off in MARWIL's default config (probably shouldn't be)
        config["evaluation_config"] = {"input": "sampler"}   # <- make sure evaluation workers have an actual Env
        config["input"] = ["[your file with pre-recorded data]"]
        num_iterations = 2000

        # Test for all frameworks.
        for _ in framework_iterator(config, frameworks="torch"):
            trainer = marwil.MARWILTrainer(config=config, env="CartPole-v0")
            for i in range(num_iterations):
                print(trainer.train())  # <- in the returned dict, look for the "evaluation" key, which holds a sub-dict with the correct rewards from the evaluation workers' envs in it.

            trainer.stop()
```

**Comment by sven1977** [comment]
@maxpumperla

**Comment by sven1977** [comment]
Will discuss internally, whether we should change the default settings of MARWIL. Also, I'm wondering, why tune would not print out the evaluation results (it prints out the "normal" workers' results, hence NaN).

**Comment by sven1977** [comment]
I'm closing this issue. Feel free to re-open in case the above solution does not work on your end.
Again, we'll make sure this becomes more user-friendly in a next version.
